pkg/sdk/testint/api_integrations_gen_integration_test.go

```diff
@@ -0,0 +1,436 @@
+			WithAwsApiProviderParams(sdk.NewAwsApiParamsRequest(sdk.ApiIntegrationAwsApiGateway, apiAwsRoleArn))
+	}
+
+	createApiIntegrationAzureRequest := func(t *testing.T) *sdk.CreateApiIntegrationRequest {
+		t.Helper()
+		id := sdk.RandomAccountObjectIdentifier()
+
+		return sdk.NewCreateApiIntegrationRequest(id, prefixes(azureAllowedPrefix), true).
+			WithAzureApiProviderParams(sdk.NewAzureApiParamsRequest(azureTenantId, azureAdApplicationId))
+	}
+
+	createApiIntegrationGoogleRequest := func(t *testing.T) *sdk.CreateApiIntegrationRequest {
+		t.Helper()
+		id := sdk.RandomAccountObjectIdentifier()
+
+		return sdk.NewCreateApiIntegrationRequest(id, prefixes(googleAllowedPrefix), true).
+			WithGoogleApiProviderParams(sdk.NewGoogleApiParamsRequest(googleAudience))
+	}
+
+	createApiIntegrationWithRequest := func(t *testing.T, request *sdk.CreateApiIntegrationRequest) *sdk.ApiIntegration {
+		t.Helper()
+		id := request.GetName()
+
+		err := client.ApiIntegrations.Create(ctx, request)
+		require.NoError(t, err)
+		t.Cleanup(cleanupApiIntegrationProvider(id))
+
+		integration, err := client.ApiIntegrations.ShowByID(ctx, id)
+		require.NoError(t, err)
+
+		return integration
+	}
+
+	createAwsApiIntegration := func(t *testing.T) *sdk.ApiIntegration {
+		t.Helper()
+		return createApiIntegrationWithRequest(t, createApiIntegrationAwsRequest(t))
+	}
+
+	createAzureApiIntegration := func(t *testing.T) *sdk.ApiIntegration {
+		t.Helper()
+		return createApiIntegrationWithRequest(t, createApiIntegrationAzureRequest(t))
+	}
+
+	createGoogleApiIntegration := func(t *testing.T) *sdk.ApiIntegration {
+		t.Helper()
+		return createApiIntegrationWithRequest(t, createApiIntegrationGoogleRequest(t))
+	}
+
+	t.Run("create api integration: aws basic", func(t *testing.T) {
+		request := createApiIntegrationAwsRequest(t)
+
+		integration := createApiIntegrationWithRequest(t, request)
+
+		assertApiIntegration(t, integration, request.GetName(), "")
+	})
+
+	t.Run("create api integration: azure basic", func(t *testing.T) {
+		request := createApiIntegrationAzureRequest(t)
+
+		integration := createApiIntegrationWithRequest(t, request)
+
+		assertApiIntegration(t, integration, request.GetName(), "")
+	})
+
+	t.Run("create api integration: google basic", func(t *testing.T) {
+		request := createApiIntegrationGoogleRequest(t)
+
+		integration := createApiIntegrationWithRequest(t, request)
+
+		assertApiIntegration(t, integration, request.GetName(), "")
+	})
+
+	t.Run("create api integration: aws more options", func(t *testing.T) {
+		request := createApiIntegrationAwsRequest(t)
+
+		request = request.
+			WithAwsApiProviderParams(request.AwsApiProviderParams.WithApiKey(sdk.String("key"))).
+			WithApiBlockedPrefixes(prefixes(awsBlockedPrefix)).
+			WithComment(sdk.String("comment"))
+
+		integration := createApiIntegrationWithRequest(t, request)
+
+		assertApiIntegration(t, integration, request.GetName(), "comment")
```

Comment:
2 newly set params could we asserted with description properties. Probably `assertApiIntegrationProperties` function should be added to test whether newly added parameters were successfully set. Same in tests below.

Reply:
or just the same thing you've added in line:181

Reply:
I can add the assertions

Reply:
Added